spec/requests/courses_spec.rb

```diff
@@ -1,19 +1,73 @@
 require 'helpers/rails_helper'
+require 'helpers/api_helper'
 
 RSpec.describe "Courses", type: :request do
   include_context "with authenticated requests"
-  let(:user) { create(:user) }
+  let(:student) { create(:user) }
+  let(:teacher) { create(:user, :teacher) }
   let!(:course) { create(:course) }
 
-  before do
```

Comment:
Ah, yes that's true. In that case, this should suffice. Or, you can split the specs into two contexts, one where a student is authenticated, and one where a teacher is authenticated.